src/parser.rs

```diff
@@ -342,6 +344,40 @@ impl<'a> Parser<'a> {
     Ok(())
   }
 
+  fn alias(&mut self, alias_name: Token<'a>) -> CompilationResult<'a, ()> {
+    // Make sure alias doesn't already exist
+    if self.aliases.contains_key(alias_name.lexeme) {
+      return Err(alias_name.error(DuplicateAlias {
+        alias: alias_name.lexeme, 
+      }))
+    }
+
+    // Make sure the next token is of kind Name and keep it
+    let next = self.tokens.next().unwrap();
+    let recipe_name;
+    match next.kind {
+      Name => recipe_name = next,
+      _ => return Err(self.unexpected_token(&next, &[Name]))
+    }
```

Comment:
The `accept` helper can make this a little bit cleaner:

```rust
// Make sure the next token is of kind Name and keep it   
let target = if let Some(name) = self.accept(Name) {      
  name.lexeme                                             
} else {                                                  
  let unexpected = self.tokens.next().unwrap();           
  return Err(self.unexpected_token(&unexpected, &[Name]));
};                                                        
```

Reply:
Done, this is better. I didn't know one could write `if let ... = ... {}` earlier, I made changes to use this pattern in a couple of other places.